src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -252,17 +252,22 @@ void UDPEndPointImplLwIP::CloseImpl()
         mUDP              = nullptr;
         mLwIPEndPointType = LwIPEndPointType::Unknown;
 
-        // In case that there is a UDPEndPointImplLwIP::LwIPReceiveUDPMessage
+        // If there is a UDPEndPointImplLwIP::LwIPReceiveUDPMessage
         // event pending in the event queue (SystemLayer::ScheduleLambda), we
         // schedule a release call to the end of the queue, to ensure that the
         // queued pointer to UDPEndPointImplLwIP is not dangling.
-        Retain();
-        CHIP_ERROR err = GetSystemLayer().ScheduleLambda([this] { Release(); });
-        if (err != CHIP_NO_ERROR)
+        if (mDelayRelease == true)
```

Comment:
not: seems a bit awkward to use `== true` on a bool. Why not `if (mDelayRelease)`